[
  {
    "package": {
      "name": "cryptography",
      "version": "40.0.1",
      "ecosystem": "PyPI",
      "commit": ""
    },
    "vulnerabilities": [
      {
        "modified": "2023-11-09T05:40:01Z",
        "published": "2023-07-14T21:31:08Z",
        "schema_version": "1.6.0",
        "id": "GHSA-cf7p-gm2m-833m",
        "aliases": [
          "CVE-2023-38325"
        ],
        "summary": "cryptography mishandles SSH certificates",
        "details": "The cryptography package before 41.0.2 for Python mishandles SSH certificates that have critical options.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "40.0.0"
                  },
                  {
                    "fixed": "41.0.2"
                  }
                ]
              }
            ],
            "versions": [
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2023/07/GHSA-cf7p-gm2m-833m/GHSA-cf7p-gm2m-833m.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N"
          }
        ],
        "references": [
          {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-38325"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/issues/9207"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/7960"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/9208"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/commit/1ca7adc97b76a9dfbd3d850628b613eb93b78fc3"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/pyca/cryptography"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/compare/41.0.1...41.0.2"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pypa/advisory-database/tree/main/vulns/cryptography/PYSEC-2023-112.yaml"
          },
          {
            "type": "WEB",
            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/NMCCTYY3CSNQBFFYYC5DAV6KATHWCUZK/"
          },
          {
            "type": "WEB",
            "url": "https://pypi.org/project/cryptography/#history"
          },
          {
            "type": "WEB",
            "url": "https://security.netapp.com/advisory/ntap-20230824-0010/"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-295"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2023-07-14T22:14:26Z",
          "nvd_published_at": "2023-07-14T20:15:09Z",
          "severity": "HIGH"
        }
      },
      {
        "modified": "2023-08-21T15:28:13Z",
        "published": "2023-07-14T20:15:00Z",
        "schema_version": "1.6.0",
        "id": "PYSEC-2023-112",
        "aliases": [
          "CVE-2023-38325"
        ],
        "details": "The cryptography package before 41.0.2 for Python mishandles SSH certificates that have critical options.",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "cryptography",
              "purl": "pkg:pypi/cryptography"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "40.0.0"
                  },
                  {
                    "fixed": "41.0.2"
                  }
                ]
              }
            ],
            "versions": [
              "40.0.0",
              "40.0.1",
              "40.0.2",
              "41.0.0",
              "41.0.1"
            ],
            "database_specific": {
              "source": "https://github.com/pypa/advisory-database/blob/main/vulns/cryptography/PYSEC-2023-112.yaml"
            }
          }
        ],
        "references": [
          {
            "type": "PACKAGE",
            "url": "https://pypi.org/project/cryptography/#history"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/compare/41.0.1...41.0.2"
          },
          {
            "type": "REPORT",
            "url": "https://github.com/pyca/cryptography/issues/9207"
          },
          {
            "type": "WEB",
            "url": "https://github.com/pyca/cryptography/pull/9208"
          }
        ]
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-cf7p-gm2m-833m",
          "PYSEC-2023-112"
        ]
      }
    ]
  }
]